core/MyGatewayTransportMQTTClient.cpp

```diff
@@ -50,8 +50,24 @@ IPAddress _subnetIp(255, 255, 255, 0);
 uint8_t _MQTT_clientMAC[] = { MY_MAC_ADDRESS };
 #endif /* End of MY_GATEWAY_ESP8266 */
 
+#if defined(MY_GATEWAY_TINYGSM)
```

Comment:
We should probably indicate that the GSM gateway only applies to MQTT GW:s either though the define or at least in the doxygen docs for the flag. If users enable the MY_GATEWAY_TINYGSM flag, and don't indicate MQTT it won't work.
The example does this, but in case people forget that flag when rolling their own gw it might cause some forum "noise".
An option could be to put a guard in MyConfig.h, along the lines of this:
```
#if defined(MY_GATEWAY_TINYGSM) && !defined(MY_GATEWAY_MQTT_CLIENT)
#error MY_GATEWAY_TINYGSM only works with MY_GATEWAY_MQTT_CLIENT
#endif
```

Reply:
Good idea. I also added a check to make sure both TX and RX pins are defined when using softserial